sequencer.c

```diff
@@ -3811,6 +3821,7 @@ static int pick_commits(struct repository *r,
 		struct todo_item *item = todo_list->items + todo_list->current;
 		const char *arg = todo_item_get_arg(todo_list, item);
 		int check_todo = 0;
+		struct object_id head;
```

Comment:
Let's move this further down into the `if (item->command <= TODO_SQUASH)` block, to restrict the scope further.